src/bls12_381/fq2.rs

```diff
@@ -54,6 +56,19 @@ impl Fq2 {
 
         t1
     }
+
+    pub fn parity(&self) -> bool {
+        self.c0.parity()
```

Comment:
See my other comment on parity. (Also, this wouldn't work, because c0 can be zero while c1 is nonzero, so the two opposite points would return the same parity value. This is why we use SORT encoding instead.)